src/openforms/registrations/contrib/json/config.py

```diff
@@ -0,0 +1,51 @@
+from typing import Required, TypedDict
+
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework import serializers
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+
+from openforms.api.fields import PrimaryKeyRelatedAsChoicesField
+from openforms.formio.api.fields import FormioVariableKeyField
+from openforms.utils.mixins import JsonSchemaSerializerMixin
+
+
+class JSONOptionsSerializer(JsonSchemaSerializerMixin, serializers.Serializer):
+    service = PrimaryKeyRelatedAsChoicesField(
+        queryset=Service.objects.filter(api_type=APITypes.orc),
+        label=_("Service"),
+        help_text=_("Which service to use."),
+        required=True,
+    )
+    relative_api_endpoint = serializers.CharField(
+        max_length=255,
+        label=_("Relative API endpoint"),
+        help_text=_(
+            "The API endpoint to send the data to (relative to the service API root)."
+        ),
+        allow_blank=True,
+        required=False,
+    )
+    form_variables = serializers.ListField(
+        child=FormioVariableKeyField(max_length=50),
```

Comment:
```suggestion
        child=FormioVariableKeyField(),
```
`key` on `FormVariable` has no length limits (it's `models.TextField`), so we can't add such constraints as it would exclude certain variables from being sent.